File: Array | Dictionary | Set/two_sum/py/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}

        for indx,num in enumerate(nums):
            difference = target - num

            val = hashmap.get(num)
            if val != None:
                return [val,indx] 

            hashmap[difference] = indx

        return [0,1]

File: Array | Dictionary | Set/two_sum/js/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const hashmap = new Map();

    for (const [indx, val] of nums.entries()) {
        const difference = target - val;

        if (hashmap.has(difference)) {
            return [hashmap.get(difference), indx];
        }

        hashmap.set(val, indx);
    }

    return [0, 1];
};
